### PR TITLE
[FLINK-13351][table-blink-planner]duplicate case ROW match in FlinkTy…

### PR DESCRIPTION
**Pull request by AjaxXu**
…peFactory.toLogicalType

## What is the purpose of the change

*This pull request remove duplicate case Row match in FlinkTypeFactory in table-blink-planner*

## Brief change log

 - *remove duplicate case Row match in FlinkTypeFactory in table-blink-planner*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala [hunk 1]
```diff
@@ -479,12 +479,6 @@ object FlinkTypeFactory {
...
         genericRelDataType.genericType
 
-      case ROW if relDataType.isInstanceOf[RelRecordType] =>
-        val recordType = relDataType.asInstanceOf[RelRecordType]
-        RowType.of(
-          recordType.getFieldList.map(_.getType).map(toLogicalType).toArray,
-          recordType.getFieldNames.toSeq.toArray)
-
       case ROW if relDataType.isInstanceOf[RelRecordType] =>
         toLogicalRowType(relDataType.asInstanceOf[RelRecordType])
```
Comment (JingsongLi) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala]: Remove cast.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala [hunk 2]
```diff
@@ -479,14 +479,7 @@ object FlinkTypeFactory {
...
 
-      case ROW if relDataType.isInstanceOf[RelRecordType] =>
-        val recordType = relDataType.asInstanceOf[RelRecordType]
-        RowType.of(
-          recordType.getFieldList.map(_.getType).map(toLogicalType).toArray,
-          recordType.getFieldNames.toSeq.toArray)
-
-      case ROW if relDataType.isInstanceOf[RelRecordType] =>
-        toLogicalRowType(relDataType.asInstanceOf[RelRecordType])
+      case ROW => toLogicalRowType(relDataType)
```
Comment (AjaxXu) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala]: Thanks @JingsongLi I have changed the code.

**Comment by AjaxXu** [comment]
@wuchong Would you mind to help me to review and merge this pr?Thanks.

**Comment by wuchong** [comment]
Travis passed: https://travis-ci.com/flink-ci/flink/builds/120314747
Will merge this.

**Comment by AjaxXu** [comment]
> LGTM.
> 
> Thanks for the contribution @AjaxXu . Please note that try to use "git rebase" instead of "git merge".

Thanks for your kind remind. I will do it later.

**Pull request closed without merging**